src/server.ts

```diff
@@ -47,100 +47,141 @@ const startServer = async () => {
+
+      request.auth = tokenData
+
+      if (error) {
+        reply.statusCode = 401
+        return reply.send({ success: false, message: error })
+      }
+    })
+
+    // Routes to work without authentication check, behind /public route
+    server.register(
+      (server, _, done) => {
+        server.post('/login', routeLogin)
+        server.get('/localisations', routeLocalisations)
+        server.get('/language/:code', routeGetLanguageFile)
+        server.get('/verify', routeVerification)
+        done()
+      },
+      { prefix: '/public' }
+    )
+    // Route for admin tasks only
+    server.register(
+      (server, _, done) => {
+        // Check isAdmin
+        server.addHook('preValidation', async (request: any, reply: FastifyReply) => {
+          if (!request.auth.isAdmin) {
+            reply.statusCode = 401
+            return reply.send({ sucess: false, message: 'Not admin user' })
+          }
+        })
+
+        server.register(lookupTableRoutes, { prefix: '/lookup-table' })
+        server.register(snapshotRoutes, { prefix: '/snapshot' })
+        server.get('/updateRowPolicies', routeUpdateRowPolicies)
+        done()
+      },
+      { prefix: '/admin' }
     )
-    // TO-DO Check for permission to access file
-    try {
-      // TO-DO: Rename file back to original for download
-      return reply.sendFile(file_path ? file_path : thumbnail_path)
-    } catch {
-      return reply.send({ success: false, message: 'Unable to retrieve file' })
-    }
-  })
-
-  server.get('/user-info', routeUserInfo)
-  server.post('/login', routeLogin)
-  server.post('/login-org', routeLoginOrg)
-  server.get('/updateRowPolicies', routeUpdateRowPolicies)
-  server.post('/create-hash', routeCreateHash)
-  server.post('/run-action', routeRunAction)
-  server.get('/verify', routeVerification)
-  server.post('/generate-pdf', routeGeneratePDF)
-  server.get('/localisations', routeLocalisations)
-  server.get('/language/:code', routeGetLanguageFile)
-  server.get('/outcomes', routeOutcomes)
-  server.get('/outcomes/table/:tableName', routeOutcomesTable)
-  server.get('/outcomes/table/:tableName/item/:id', routeOutcomesDetail)
-
-  // File upload endpoint
-  server.post('/upload', async function (request: any, reply) {
-    // TO-DO: Authentication
-    const data = await request.files()
-    const fileData = await saveFiles(data, objectKeysToSnakeCase(request.query))
-    reply.send({ success: true, fileData })
-  })
-
-  server.register(lookupTableRoutes, { prefix: '/lookup-table' })
 
-  // Snapshot routes
-  server.register(snapshotRoutes, { prefix: '/snapshot' })
+    // File download endpoint (get by unique ID)
+    server.get('/file', async function (request: any, reply: any) {
+      const { uid, thumbnail } = request.query
+      const { original_filename, file_path, thumbnail_path } = await getFilePath(
+        uid,
+        thumbnail === 'true'
+      )
+      // TO-DO Check for permission to access file
```

Comment:
@andreievg -- do you think uniqueID is enough to keep files accessible by people who shouldn't, or do we need to add some user-based checks?

It's a bit tricky if we do, cos some files will need to be widely accessible (such as org logos, which show up in Outcomes), but some should only be accessed by applicants and reviewer's reviewing their applications.